db/blip_handler.go

```diff
@@ -931,7 +931,7 @@ func (bh *blipHandler) handleProveAttachment(rq *blip.Message) error {
 		return base.HTTPErrorf(http.StatusBadRequest, "no digest sent with proveAttachment")
 	}
 
-	attData, err := bh.db.GetAttachment(AttachmentKey(digest))
+	attData, err := bh.db.GetAttachment(base.AttPrefix + digest)
```

Comment:
If we get an error here, check bh.clientType to see if it's ISGR.  If it's ISGR, return the 'AttachmentNotFound' error here.

Reply:
Sure, that makes sense. Added changes to check for bh.clientType == BLIPClientTypeSGR2 and return ErrAttachmentNotFound.